ddtrace/internal/module.py

```diff
@@ -189,7 +193,22 @@ def _create_module(self, spec):
         return self.loader.create_module(spec)
 
     def _exec_module(self, module):
-        self.loader.exec_module(module)
+        # Collect all the transformers registered by other ModuleWatchdog
+        # instances so that we can run them all in order.
+        pre_exec_hooks = []
+
+        for _ in sys.meta_path:
+            if isinstance(_, ModuleWatchdog):
+                for cond, hook in _._pre_exec_module_hooks:
+                    if isinstance(cond, str) and cond == module.__name__ or cond(module.__name__):
+                        pre_exec_hooks.append(hook)
+
+        if pre_exec_hooks:
+            for hook in pre_exec_hooks:
+                hook(self, module)
```

Comment:
maybe we can wrap this around `try...except` and raise an exception if we couldn't run the hook succesfully. Then we use this signal to call the default loader instead?

Reply:
Hmmm... after giving it a thought... what if the exception is there when you execute a module? ImportError, RuntimeError, OSError... then we'd execute it twice. I believe the hook should capture its exceptions so the original ones are still propagated